src/github.com/matrix-org/dendrite/clientapi/producers/roomserver.go

```diff
@@ -45,22 +45,60 @@ func NewRoomserverProducer(kafkaURIs []string, topic string) (*RoomserverProduce
 func (c *RoomserverProducer) SendEvents(events []gomatrixserverlib.Event) error {
 	eventIDs := make([]string, len(events))
 	ires := make([]api.InputRoomEvent, len(events))
-	for i := range events {
-		var authEventIDs []string
-		for _, ref := range events[i].AuthEvents() {
-			authEventIDs = append(authEventIDs, ref.EventID)
-		}
-		ire := api.InputRoomEvent{
+	for i, event := range events {
+		ires[i] = api.InputRoomEvent{
 			Kind:         api.KindNew,
-			Event:        events[i].JSON(),
-			AuthEventIDs: authEventIDs,
+			Event:        event.JSON(),
+			AuthEventIDs: authEventIDs(event),
 		}
-		ires[i] = ire
-		eventIDs[i] = events[i].EventID()
+		eventIDs[i] = event.EventID()
 	}
 	return c.SendInputRoomEvents(ires, eventIDs)
 }
 
+// SendEventWithState writes an event with KindNew to the roomserver input log
+// preceeded by the state at the event as KindOutlier.
```

Comment:
I would probably drop that word since whilst it generally means "before" it can also be used as "after" in some contexts.